src/services/ActivityService.ts

```diff
@@ -110,4 +150,54 @@ export class ActivityService {
                 throw new HTTPResponse(error.statusCode, { error: `${error.code}: ${error.message} At: ${error.hostname} - ${error.region} Request id: ${error.requestId}` });
             });
     }
+
+    /**
+     * Updates an activity in the database.
+     * @param activity - the payload containing the activity
+     * @returns Promise - void
+     */
+    public async updateActivity(activities: IActivity[]): Promise<void> {
+        const activitiesList: any[] = [];
+        for (const each of activities) {
+            // Payload validation
+            const validation: Joi.ValidationResult<IActivity> = Joi.validate(each, ActivityUpdateSchema);
+            if (validation.error) {
+                const error: string = validation.error.details[0].message;
+
+                throw new HTTPResponse(400, {error});
+            }
+            await this.dbClient.get({id: each.id})
+                .then(async (result: DocumentClient.GetItemOutput): Promise<void> => {
+                    // Result checks
+                    if (result.Item === undefined) {
+                        throw new HTTPResponse(404, {error: HTTPRESPONSE.NOT_EXIST});
+                    }
+
+                    const dbActivity: IActivity = result.Item as IActivity;
+
+                    // Assign the waitReasons
+                    Object.assign(dbActivity, {waitReason: each.waitReason});
+                    // Assign the notes
+                    Object.assign(dbActivity, {notes: each.notes});
+                    activitiesList.push(dbActivity);
+                    console.log(`Updating each activity: ${dbActivity.id} with waitReason.`);
```

Comment:
Is there any reason why you want to log this?

Reply:
Was more for local tests. Will remove them.